.misc/Fudge.ps1

```diff
@@ -0,0 +1,448 @@
+                    -Parameters $Parameters -Arguments $Arguments -Dev:$Dev -DevOnly:$DevOnly -Adhoc:$Adhoc
+
+                Invoke-ChocolateyAction -Action 'install' -Key $Key -Source $Source -Config $config -LocalList $localList `
+                    -Parameters $Parameters -Arguments $Arguments -Dev:$Dev -DevOnly:$DevOnly -Adhoc:$Adhoc
+            }
+
+        {($_ -ieq 'pack')}
+            {
+                Invoke-ChocolateyAction -Action 'pack' -Key $Key -Config $config
+            }
+
+        {($_ -ieq 'list')}
+            {
+                Invoke-FudgeLocalDetails -Config $config -Key $Key -LocalList $localList -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        {($_ -ieq 'search')}
+            {
+                Invoke-Search -Key $Key -Limit $Limit -Source $Source -LocalList $localList
+            }
+
+        {($_ -ieq 'new')}
+            {
+                New-Fudgefile -Path $FudgefilePath -Key $Key -LocalList $localList -Install:$Install -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        {($_ -ieq 'renew')}
+            {
+                Restore-Fudgefile -Path $FudgefilePath -Key $Key -LocalList $localList -Install:$Install -Uninstall:$Uninstall -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        {($_ -ieq 'delete')}
+            {
+                Remove-Fudgefile -Path $FudgefilePath -Uninstall:$Uninstall -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        {($_ -ieq 'prune')}
+            {
+                Invoke-FudgePrune -Config $config -LocalList $localList -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        {($_ -ieq 'clean')}
+            {
+                Invoke-FudgeClean -LocalList $localList
+            }
+
+        {($_ -ieq 'add')}
+            {
+                Invoke-FudgeAdd -Path $FudgefilePath -Key $Key -Source $Source -Config $config -LocalList $localList `
+                    -Parameters $Parameters -Arguments $Arguments -Dev:$Dev -Install:$Install
+            }
+
+        {($_ -ieq 'remove')}
+            {
+                Invoke-FudgeRemove -Path $FudgefilePath -Key $Key -Config $config -LocalList $localList `
+                    -Parameters $Parameters -Arguments $Arguments -Dev:$Dev -Uninstall:$Uninstall
+            }
+
+        {($_ -ieq 'which')}
+            {
+                Invoke-FudgeWhich -Key $Key
+            }
+
+        {($_ -ieq 'rebuild')}
+            {
+                Invoke-FudgeClean -LocalList $localList
+                Invoke-ChocolateyAction -Action 'install' -Key $Key -Source $Source -Config $config -Dev:$Dev -DevOnly:$DevOnly
+            }
+
+        default
+            {
+                Write-Fail "Action not recognised: $($_)"
+            }
+    }
+}
+finally
+{
+    # output duration, and cleanup
+    Write-Details "`nDuration: $(([DateTime]::UtcNow - $timer).ToString())"
+    Remove-Module -Name 'FudgeTools' -ErrorAction SilentlyContinue | Out-Null
+}
```

Comment:
EOL at EOF?

Reply:
this is code imported from https://github.com/Badgerati/Fudge ; raise issues or PRs there :P